### PR TITLE
Library Reorganization

### PR DESCRIPTION
**Pull request by camagallen**
## Overview
This PR contains major changes to the structure of the library intended to make it more intuitive. Documentation was updated accordingly.

### Important Notes to Reviewers
- **Do not waste time reviewing the jupyter notebooks.** There will be an upcoming PR dedicated to validating our notebooks. **Please do not review any jupyter notebooks in conjunction with this PR**
- Due to the type of updates in this PR, I suggest starting the review by pulling and the "reorg" branch and considering current changes locally before looking line-by-line (if that is not your current workflow).

### Major Changes
- fairMLHealth modules were renamed to clarify scope and purpose. This will facilitate adding new features in upcoming PRs: 
    - **reports.py** is now named **measure.py**
    - **model_comparison** is now named **report.py**
    - utils.py is now split into stat_utils.py and __utils.py
        - __utils.py contains back-end functions for which validation is assumed to have been run
        - stat_utils.py contains supplemental functions that may be useful in statistical analysis or data munging
- newly-renamed measure.py has also been reorganized so that functions are in alphabetical order and names are more intuitive. (this should have been a separate PR. apologies in advance).
- load_mimic_data.py is now "__mimic_data.py"
- the "tutorials_and_examples" folder has been renamed "examples_and_tutorials" so that it will be easier to find (e.g. seen adjacent to the "docs" folder)
- all validation functions are now contained within __validation.py

Resolves: [AB#12353](https://dev.azure.com/kensci/91234890-e1c9-4679-971f-09afd8516db1/_workitems/edit/12353)

## How was this update tested?
A new test script was written to test the new stat_utils.py module. Pytest ran and completed successfully.

## Checklist
- [x] I have read the [Contributing](https://github.com/KenSciResearch/docs/code_contributions/CONTRIBUTING.md) guidelines
- [ ] I performed a self-review of my own code
- [x] The CHANGELOG and version.json have been updated
- [x] Documentation has been updated to reflect the changes
    - [x] The code has clear and helpful comments, particularly in hard-to-understand areas
    - [x] READMEs, Tutorials/Examples, and links have been updated as appropriate
- [x] Tests have been added to prove that the fix is effective or that the feature works
- [x] Pytest was run; new and existing unit tests pass locally with my changes
- [x] I have made note of any breakages or dependencies (if applicable)
- [x] Any dependent changes have been merged and published in downstream modules
- [x] These changes generate no new warnings

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,22 +4,28 @@ Tools and tutorials for evaluation of fairness and bias in healthcare applicatio
...
+    - Tutorials for measuring and analyzing fairness as it applies to machine learning
+    - Examples for using the templates and tools
+
+- ### [FairMLHealth](fairmlhealth/README.md)
+    - **Measure**:
+        - Stratified tables useful in data and model validation
+        - Tools for detailed analysis across multiple indicators (e.g. when location of bias is undetermined)
+    - **Report**:
+        - Tools stylized for inclusion in publications and analytical reports
+        - Tools for monitoring and reporting on operationalized models
```
Comment (kenviku) [inline README.md:22]: I think we can remove `monitoring`?

File: README.md [hunk 2]
```diff
@@ -70,13 +76,13 @@ model_dict = {'model_1': model_1, 'model_2': model_2}
 ### Measuring
 The primary feature of this library is the model comparison tool. The current version supports assessment of binary prediction models through use of the measure_models and compare_models functions.
 
-Measure_model is designed to generate a report of multiple fairness metrics for a single model. Here it is shown wrapped in a "flag" function to emphasize values that are outside of the "fair" range.
+Measure_model is designed to generate an analysis table of multiple fairness metrics for a single model. Here it is shown wrapped in a "flag" function to emphasize values that are outside of the "fair" range.
 
 ``` python
 # Generate a pandas dataframe of measures
-fairness_measures = fhmc.measure_model(X_test, y_test, prtc_attr, model_1)
+fairness_measures = report.measure_model(X_test, y_test, prtc_attr, model_1)
```
Comment (kenviku) [inline README.md:83]: A bit counter-intuitive. There is a `measure.py` but then one calls `measure_model` in `report.py`?

Comment (camagallen) [reply README.md:83]: Hmm. This is a good point. Will address in a later PR

**Review by kenviku — APPROVED** [review]
LGTM

**Pull request merged**